rust/cas_client/src/data_transport.rs

```diff
@@ -102,18 +108,36 @@ impl DataTransport {
     /// creates the DataTransport instance for the H2 connection using
     /// CasConnectionConfig info, and additional port
     pub async fn from_config(cas_connection_config: CasConnectionConfig) -> Result<Self> {
-        debug!(
+        info!(
             "Attempting to make HTTP connection with {}",
             cas_connection_config.endpoint
         );
-        let h2_client = Client::builder()
+        let mut builder = Client::builder(TokioExecutor::new());
+        builder
+            .timer(TokioTimer::new())
             .pool_idle_timeout(Duration::from_secs(HTTP2_POOL_IDLE_TIMEOUT_SECS))
             .http2_keep_alive_interval(Duration::from_millis(HTTP2_KEEPALIVE_MILLIS))
             .http2_initial_connection_window_size(HTTP2_WINDOW_SIZE)
             .http2_initial_stream_window_size(HTTP2_WINDOW_SIZE)
-            .http2_only(true)
-            .build_http();
-
+            .http2_only(true);
+        let connector = if let Some(root_ca) = &cas_connection_config.root_ca {
+            info!("connector with custom root ca");
+            let ca = Certificate::from_pem(root_ca.as_bytes())?;
+            let tls_connector = native_tls::TlsConnector::builder()
+                .danger_accept_invalid_certs(true)
```

Comment:
nit: this is such a scary function name that it deserves a comment explaining why ok

Reply:
I am having trouble understanding why we need this. Shouldn't `add_root_certificate` update the trust store with our root certificate, thereby making any requests to the CAS server valid?

Reply:
After some investigation I found the following:

- Setting danger_accept_invalid_certs will make encryption work regardless of specifying the certificate.
- Setting just the certificate (`add_root_certificate`) without setting the danger_accept_invalid_certs will result in a Connect error which is bizarre, telling me I'm using it wrong or it's not actually working.
  - `Error { kind: Connect, source: Some(Error { code: -67843, message: "The certificate was not trusted." }) }`

Reply:
switching to using rustls, over native_tls, solves the issue.